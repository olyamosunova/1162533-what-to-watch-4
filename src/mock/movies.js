const PromoMovie = {
  TITLE: `The Grand Budapest Hotel`,
  GENRE: `Drama`,
  RELEASE_DATE: 2014,
  POSTER: `the-grand-budapest-hotel-poster.jpg`,
  COVER: `bg-the-grand-budapest-hotel.jpg`,
};

const MoviesList = [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody.jpg`,
  },
  {
    id: 3,
    title: `Macbeth`,
    poster: `macbeth.jpg`,
  },
  {
    id: 4,
    title: `Aviator`,
    poster: `aviator.jpg`,
  },
  {
    id: 5,
    title: `We need to talk about Kevin`,
    poster: `we-need-to-talk-about-kevin.jpg`,
  },
  {
    id: 6,
    title: `What We Do in the Shadows`,
    poster: `what-we-do-in-the-shadows.jpg`,
  },
  {
    id: 7,
    title: `Revenant`,
    poster: `revenant.jpg`,
  },
  {
    id: 8,
    title: `Johnny English`,
    poster: `johnny-english.jpg`,
  },
];

const MovieDescriptions = {
  TITLE: `The Grand Budapest Hotel`,
  GENRE: `Drama`,
  RELEASE_DATE: 2014,
  POSTER: `the-grand-budapest-hotel-poster.jpg`,
  COVER: `bg-the-grand-budapest-hotel.jpg`,
  RATING: 8.9,
  RATING_LEVEL: `Very good`,
  RATING_COUNT: 240,
  DESCRIPTION: [
    `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  ],
  DIRECTOR: `Wes Andreson`,
  STARRING: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other`,
};

export {PromoMovie, MoviesList, MovieDescriptions};
