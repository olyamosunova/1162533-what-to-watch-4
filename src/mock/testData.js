const genres = [`Drama`, `Comedy`, `Western`];

const movie = {
  promoMovie: {
    id: 1,
    title: `Macbeth`,
    poster: `macbeth.jpg`,
    cover: `macbeth.jpg`,
    genre: `Drama`,
    releaseDate: 2015,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  backgroundColor: ``,
  previewImage: ``,
  videoLink: ``,
  rating: 7.1,
  ratingLevel: `Very good`,
  ratingCount: 240,
  runTime: 120,
  description: ``,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
    `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
};

const Movies = [
  {
    promoMovie: {
      id: 1,
      title: `Fantastic Beasts: The Crimes of Grindelwald`,
      poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      cover: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      genre: `Fantasy`,
      releaseDate: 2018,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 6.6,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 2,
      title: `Bohemian Rhapsody`,
      poster: `bohemian-rhapsody.jpg`,
      cover: `bohemian-rhapsody.jpg`,
      genre: `Music`,
      releaseDate: 2018,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 8.0,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 3,
      title: `Macbeth`,
      poster: `macbeth.jpg`,
      cover: `macbeth.jpg`,
      genre: `Drama`,
      releaseDate: 2015,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.1,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 4,
      title: `Aviator`,
      poster: `aviator.jpg`,
      cover: `aviator.jpg`,
      genre: `Drama`,
      releaseDate: 2014,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.6,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 5,
      title: `We need to talk about Kevin`,
      poster: `we-need-to-talk-about-kevin.jpg`,
      cover: `we-need-to-talk-about-kevin.jpg`,
      genre: `Drama`,
      releaseDate: 2011,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.5,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 6,
      title: `What We Do in the Shadows`,
      poster: `what-we-do-in-the-shadows.jpg`,
      cover: `what-we-do-in-the-shadows.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 7,
      title: `Revenant`,
      poster: `revenant.jpg`,
      cover: `revenant.jpg`,
      genre: `Vestern`,
      releaseDate: 2015,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,

    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.8,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 8,
      title: `Johnny English`,
      poster: `johnny-english.jpg`,
      cover: `johnny-english.jpg`,
      genre: `Comedy`,
      releaseDate: 2003,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 6.2,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 9,
      title: `Fantastic Beasts: The Crimes of Grindelwald`,
      poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      cover: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      genre: `Fantasy`,
      releaseDate: 2018,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 6.6,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 10,
      title: `Bohemian Rhapsody`,
      poster: `bohemian-rhapsody.jpg`,
      cover: `bohemian-rhapsody.jpg`,
      genre: `Music`,
      releaseDate: 2018,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 8.0,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
];

const reviews = [{
  id: 1,
  message: ``,
  rating: 8.9,
  author: ``,
  date: ``,
}];

const userData = {
  id: 0,
  email: `vesan@mail.ru`,
  name: `Lola`,
  avatarUrl: ``,
};

export {movie, Movies, reviews, genres, userData};
