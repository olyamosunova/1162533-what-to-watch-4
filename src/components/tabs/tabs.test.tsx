import * as React from "react";
import * as renderer from 'react-test-renderer';
import {emptyFunction} from "../../mock/testData";
import Tabs from "./tabs";

it(`Should Tabs render correctly`, () => {
  const tree = renderer
    .create(<Tabs
      activeTab={`Overview`}
      onTabClick={emptyFunction}
    />, {
      createNodeMock: ()=>{
        return {};
      }
    })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
