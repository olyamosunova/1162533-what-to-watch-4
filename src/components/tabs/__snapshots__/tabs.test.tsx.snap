// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Tabs render correctly 1`] = `
<nav
  className="movie-nav movie-card__nav"
>
  <ul
    className="movie-nav__list"
  >
    <li
      className="movie-nav__item movie-nav__item--active"
      onClick={[Function]}
    >
      <a
        className="movie-nav__link"
        href="#"
      >
        Overview
      </a>
    </li>
    <li
      className="movie-nav__item "
      onClick={[Function]}
    >
      <a
        className="movie-nav__link"
        href="#"
      >
        Details
      </a>
    </li>
    <li
      className="movie-nav__item "
      onClick={[Function]}
    >
      <a
        className="movie-nav__link"
        href="#"
      >
        Reviews
      </a>
    </li>
  </ul>
</nav>
`;
